flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -464,28 +465,38 @@ private void callExplain(SqlCommandCall cmdCall) {
 	}
 
 	private void callSelect(SqlCommandCall cmdCall) {
+		final long start = System.currentTimeMillis();
 		final ResultDescriptor resultDesc;
 		try {
 			resultDesc = executor.executeQuery(context, cmdCall.operands[0]);
 		} catch (SqlExecutionException e) {
 			printExecutionException(e);
 			return;
 		}
-		final CliResultView view;
-		if (resultDesc.isMaterialized()) {
-			view = new CliTableResultView(this, resultDesc);
-		} else {
-			view = new CliChangelogResultView(this, resultDesc);
-		}
+		final long end = System.currentTimeMillis();
```

Comment:
How can a user cancel the submitted query?

Reply:
User can wait a timeout(setting by max-query-timeout-ms) or just type CTRL + C to terminate the submitted query.